src/phase1.hpp

```diff
@@ -601,31 +602,27 @@ std::vector<uint64_t> RunPhase1(
         0,
         globals.stripe_size);
 
-    // The max value our input (x), can take. A proof of space is 64 of these x values.
-    uint64_t max_value = ((uint64_t)1 << (k)) - 1;
-
     // These are used for sorting on disk. The sort on disk code needs to know how
     // many elements are in each bucket.
     std::vector<uint64_t> table_sizes = std::vector<uint64_t>(8, 0);
 
     // Instead of computing f1(1), f1(2), etc, for each x, we compute them in batches
     // to increase CPU efficency.
-    for (uint64_t lp = 0; lp <= (((uint64_t)1) << (k - kBatchSizes)); lp++) {
-        // For each pair x, y in the batch
-        for (auto kv : f1.CalculateBuckets(Bits(x, k), 2 << (kBatchSizes - 1))) {
-            Bits to_write = std::get<0>(kv) + std::get<1>(kv);
+    for (uint64_t lp = 0; lp < (uint64_t)1 << (k - kBatchSizes); lp++) {
+        f1.CalculateBuckets(x, 1 << kBatchSizes, f1_entries);
+        for (int i = 0; i < 1 << kBatchSizes; i++) {
```

Comment:
should use uint_t type

Reply:
I think `int` is reasonable. We don't want (or need) modulo-2 arithmetic on this variable

Reply:
oops sorry i meant we should always specify the width, ala uint64_t, instead of using generic int which is platform specific

Reply:
makes sense. my concern, generally, with unsigned types is that all arithmetic is effectively two operations. the arithmetic operation as well as an `& 0xffffffff` (or whatever width the variable has). As long as your hardware has arithmetic operators with a built-in width (say, ways of referring to smaller parts of registers as x86 does) that's fine. But if it doesn't, if all registers are 64 bits wide, then the compiler will need to lower it to two instructions, which is wasteful where we don't rely on modulo-2 arithmetics